src/libraries/Common/src/System/Security/Cryptography/Helpers.cs

```diff
@@ -142,6 +143,61 @@ Oids.EcPublicKey or
             };
         }
 
+        internal static string? GetOidFromHashAlgorithm(HashAlgorithmName algName)
+        {
+            if (algName == HashAlgorithmName.MD5)
+                return Oids.Md5;
+            if (algName == HashAlgorithmName.SHA1)
+                return Oids.Sha1;
+            if (algName == HashAlgorithmName.SHA256)
+                return Oids.Sha256;
+            if (algName == HashAlgorithmName.SHA384)
+                return Oids.Sha384;
+            if (algName == HashAlgorithmName.SHA512)
+                return Oids.Sha512;
+#if NET8_0_OR_GREATER
+            if (algName == HashAlgorithmName.SHA3_256)
+                return Oids.Sha3_256;
+            if (algName == HashAlgorithmName.SHA3_384)
+                return Oids.Sha3_384;
+            if (algName == HashAlgorithmName.SHA3_512)
+                return Oids.Sha3_512;
+
+            // SHAKE is parametrized by the output length, but only the 256-bit (shake-128) and
+            // 512-bit (shake-256) variants have an assigned OID.
+            // TODO add HashAlgorithmName.SHAKE256 and SHAKE512
```

Comment:
We intentionally left off Shake128 and Shake256 from HashAlgorithmName. I don't think that should change. It makes a number of things weird in other places that accept a `HashAlgorithmName` like `IncrementalHash`.